lib/compress.js

```diff
@@ -4652,18 +4652,21 @@ merge(Compressor.prototype, {
             });
         }
         // x ? y(a) : y(b) --> y(x ? a : b)
+        var arg_index;
         if (consequent instanceof AST_Call
             && alternative.TYPE === consequent.TYPE
```

Comment:
Doesn't the no call arguments case have to tested explicitly now?
```patch
+            && consequent.args.length > 0
```
I didn't see a test with matching side-effect-free call expressions with no arguments:
```js
    x ? (y || x)() : (y || x)();
```

Reply:
> Doesn't the no call arguments case have to tested explicitly now?

Should be taken care of by https://github.com/mishoo/UglifyJS2/blob/d21cb84696f47f5cba25c88527d5575b7954191a/lib/compress.js#L4681

> I didn't see a test with matching side-effect-free call expressions with no arguments:

Added in https://github.com/mishoo/UglifyJS2/pull/2562/commits/571ce3529e1202ba826ed2612ad7be6c9e7678a2

Reply:
Cool - thanks.

Reply:
... but yeah, checking for `consequent.args.length > 0` should improve performance as this is a rare path - done in https://github.com/mishoo/UglifyJS2/pull/2562/commits/4f5d12c573341821b2f198a5aa0bf5f1996d437d

Reply:
Anything to avoid `equivalent_to()` just to keep you happy :wink: